lua/telescope/make_entry.lua

```diff
@@ -894,6 +901,71 @@ function make_entry.gen_from_ctags(opts)
   end
 end
 
+function make_entry.gen_from_lsp_diagnostics(opts)
+  opts = opts or {}
+  opts.tail_path = utils.get_default(opts.tail_path, true)
+
+  local displayer = entry_display.create {
+    separator = "▏",
+    items = {
+    -- slightly increased width
+      { width = 11 },
+      { width = 48 },
+      { remaining = true }
+    }
+  }
+
+  local make_display = function(entry)
+    local filename
+    if not opts.hide_filename then
+      filename = entry.filename
+      if opts.tail_path then
+        filename = utils.path_tail(filename)
+      elseif opts.shorten_path then
+        filename = utils.path_shorten(filename)
+      end
+    end
+
+    -- add styling of entries
+    local sign = vim.trim(vim.fn.sign_getdefined("LspDiagnosticsSign" .. lsp_type_diagnostic[entry.type])[1].text)
```

Comment:
This should probably only be done once, rather than every time we do the display.